src/EnergyPlus/CommandLineInterface.cc

```diff
@@ -188,15 +188,27 @@ namespace CommandLineInterface {
 
         state.dataGlobal->outputEpJSONConversionOnly = opt.isSet("--convert-only");
 
+        bool eplusRunningViaAPI = state.dataGlobal->eplusRunningViaAPI;
+
         // Process standard arguments
         if (opt.isSet("-h")) {
             DisplayString(state, usage);
-            exit(EXIT_SUCCESS);
+            if (eplusRunningViaAPI) {
+                // we need another return code to let runEnergyPlusAsLibrary know it should not try to run anything
+                return static_cast<int>(ReturnCodes::SuccessButHelper);
+            } else {
+                exit(EXIT_SUCCESS);
+            }
```

Comment:
This is the gist. If eplusRunningViaAPI, we **return** instead of calling `exit`

Reply:
I'm wary of changing the return type of ProcessArgs to be an enum, so not doing it... That's what I static_cast to int.